Upgrade ffmpeg features and dependencies

Jesse0Michael opened the pull request:
Add the ffmpeg patches that are used in Faithlife products.

Upgade the Failthlife ffmpeg build to include a few additional enabled features. Add openssl, opus, and extended vp8/vp9 support.

The extended vorbis support requires libvpx5 which is available from debian's buster libraries.

Commit:
Add FFMPEG build patches.

Add the ffmpeg patches that are used in Faithlife products.

Commit:
Update ffmpeg features and dependencies.

Upgade the failthlife ffmpeg build to include a few additional enabled features. Add openssl, opus, and extended vp8/vp9 support.

The extended vorbis support requires libvpx5 which is available from debian's buster libraries.

Jesse0Michael commented on Dockerfile:
```diff
@@ -1,25 +1,33 @@
-FROM debian:9 AS builder
+FROM buildpack-deps:stretch AS builder
```
Changed the base build to buildpack-deps:stretch to pull in open ssl support. 
If we would rather leave it at debian:9 and pull in those dependencies ourself, we can try to do that as well.

ddunkin replied:
That includes a bunch of unrelated packages. I'd rather just include what we need.

ddunkin commented on Dockerfile:
```diff
@@ -1,25 +1,33 @@
-    wget \
-    yasm \
-    zlib1g-dev
+  autoconf \
```
Lost intentional indentation.

ddunkin commented on Dockerfile:
```diff
@@ -1,25 +1,33 @@
+RUN echo "deb http://deb.debian.org/debian buster main non-free" >> /etc/apt/sources.list
+
+RUN apt-get update \
+  && apt-get install -y \
```
Let's put the `&&` at the end of the line.

ddunkin commented on Dockerfile:
```diff
@@ -29,21 +37,26 @@ RUN wget -q http://ffmpeg.org/releases/ffmpeg-${FFMPEG_VERSION}.tar.bz2 && \
-    --enable-libx265 \
-    --enable-nonfree
+  --prefix=/opt/ffmpeg \
+  --pkg-config-flags="--static" \
```
What does this do?

ddunkin commented on Dockerfile:
```diff
@@ -29,21 +37,26 @@ RUN wget -q http://ffmpeg.org/releases/ffmpeg-${FFMPEG_VERSION}.tar.bz2 && \
+  --enable-libvpx \
+  --enable-libx264 \
+  --enable-libx265 \
+  --enable-openssl \
```
Out of order.

ddunkin commented on deb-control:
```diff
@@ -1,5 +1,5 @@
 Version: @VERSION@-@BUILD@
 Architecture: amd64
-Depends: libfdk-aac1, libmp3lame0, libtheora0, libvorbis0a, libvorbisenc2, libvorbisfile3, libvpx4, libx264-148, libx265-95
+Depends: libfdk-aac1, libmp3lame0, libtheora0, libvorbis0a, libvorbisenc2, libvorbisfile3, libvpx5, libx264-148, libx265-95, libcurl3, libopus0, libnuma1
```
Sort these alphabetically. Should this also include openssl?

ddunkin commented:
We should probably build separate packages for stretch and buster, each with the dependencies from their own versions.

Commit:
Separate stretch and buster build dependencies.

Create two separate builds, one using debian 9 'stretch' dependencies and one using debian 10 'buster' dependencies.

Commit:
Update readme for separate build dependencies.

Update readme to reflect the stretch and buster build dependent images.
